src/main/java/htsjdk/samtools/util/zip/DeflaterFactory.java

```diff
@@ -43,36 +43,21 @@
  */
 public class DeflaterFactory {
 
-    private static Constructor<IntelDeflater> intelDeflaterConstructor;
+    private static final boolean isUsingIntelDeflater;
 
     static {
-        try {
-            if (Defaults.TRY_USE_INTEL_DEFLATER) {
-                final Class<IntelDeflater> clazz = (Class<IntelDeflater>) Class.forName("htsjdk.samtools.util.zip.IntelDeflater");
-                intelDeflaterConstructor = clazz.getConstructor(Integer.TYPE, Boolean.TYPE);
-            }
-        } catch (ClassNotFoundException e) {
-            intelDeflaterConstructor = null;
-        } catch (NoSuchMethodException e) {
-            intelDeflaterConstructor = null;
-        } catch (UnsatisfiedLinkError e) {
-            intelDeflaterConstructor = null;
-        }
+        isUsingIntelDeflater = Defaults.TRY_USE_INTEL_DEFLATER && (new IntelDeflater().load());
```

Comment:
I'm personally of the opinion that as of Java 7/8+ the utility of the Intel deflator is questionable.  In the vast majority of my uses of HTSJDK I'm either writing at a decent compression level (where intel deflator doesn't really outperform modern zlib) or streaming between processes with compression level 0.

While intel-glk is currently small, it may grow, and it introduces some new transitive dependencies.  My preference in these cases is to (in my projects) exclude the transitive dependencies for features I don't use in order to keep my JARs as small as possible.

The code was already here to instantiate this reflectively, so that if the Intel code isn't there (or fails to load cleanly for any reason) things could continue to function.  I'm not suggesting this be done in all downstream projects, just in HTSJDK as it was until this commit.